docs/specs/sdk.md

```diff
@@ -0,0 +1,180 @@
+It is more preferable to have ephemeral refresh token that are only stored in memory.
+
+The SDK can be configured to store refresh token in memory only.
+
+## Logout
+
+Logout is provided by the following API
+
+```typescript
+/**
+ * This function can be called when the user has logged in.
+ *
+ * Depending on whether biometric authentication is enabled or not,
+ * this function behaves differently.
+ *
+ * When biometric authentication is disabled,
+ * calling this function will allow logging in as a different user.
+ *
+ * When biometric authentication is enabled,
+ * calling this function will ONLY terminate the current session.
+ * The last logged in user will NOT be forgotten.
+ * Logging in a different user will be disallowed in the web UI.
+ * If switching different user is desired,
+ * disableBiometric() must be called.
+ *
+ */
+function logout(): Promise<void>;
+```
+
+## Biometric authentication
+
+Biometric authentication is provided by the following API
+
+```typescript
+/**
+ * This function can be called at anytime.
+ * This function reports whether biometric authentication is available.
+ */
+function isBiometricAvailable(): Promise<boolean>;
+
+/**
+ * This function can be called at anytime.
+ * This function returns the user ID of last logged in user.
+ * The data is stored with kSecAttrAccessibleWhenUnlockedThisDeviceOnly,
+ * therefore the data is not included in backup, nor transfered to new device.
+ */
+function getLastUserID(): Promise<string | null>;
+
+/**
+ * This function can be called at anytime.
+ * This function tells if the last logged in user has enabled biometric authentication.
+ * The data is stored with kSecAttrAccessibleWhenUnlockedThisDeviceOnly,
+ * therefore the data is not included in backup, nor transfered to new device.
+ */
+function isBiometricEnabled(): Promise<boolean>;
+
+/**
+ * This function can be called when the user has logged in.
+ * This function WILL trigger biometric authentication if necessary.
+ *
+ * 1. An UUID is generated as kid.
+ * 2. A key pair is generated and stored in the keychain.
+ * 3. A challenge is requested from the server.
+ * 4. A Biometric identity is added to the user.
+ *
+ * The key pair is stored with kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly and kSecAccessControlBiometryAny,
+ * therefore the data is not included in backup, nor transfered to new device,
+ * biometric authentication or password code is prompted if necessary.
+ */
+function enableBiometric(): Promise<void>;
+
+/**
+ * This function can be called when the user has logged in, or at anytime if an option is specified.
+ * This function WILL trigger biometric authentication if necessary.
+ *
+ * The following steps are performed:
+ *
+ * 1. Call /api/identity/biometric/remove
+ * 2. If the error is identity not found, the error is always ignored.
+ * 3. If `force` is specified, the error is ignored, otherwise the error is thrown.
+ * 4. The key pair is deleted from the keychain.
```

Comment:
Would like to know, do we know if step 4 requires face/touch id? If it requires face/touch id. In one scenario, developer use this api, the app asks for face id in the UI but user cancel it. I think the user may assume nothing should be deleted?

1. Need check if delete require face/touch id, if it is not needed. I think this flow is fine.
2. If it requires face/touch id, maybe better change the order of flow to delete the key pair first and the kid should be stored in somewhere else so we can perform `/api/identity/biometric/remove` in next step? But this is a bit awkward that if key pair is removed successfully but remove identity is failed. What should be returned in `isBiometricEnabled`, seems false? But the identity will be there forever and no one can access it.

Thought or any other idea?